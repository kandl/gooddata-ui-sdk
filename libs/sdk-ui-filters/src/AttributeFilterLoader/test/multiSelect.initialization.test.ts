// (C) 2019-2022 GoodData Corporation
import { newTestAttributeFilterHandler } from "./fixtures";
import { waitForAsync } from "./testUtils";

describe("MultiSelectAttributeFilterHandler", () => {
    it("should trigger init success callback", async () => {
        const onInicStart = jest.fn();
        const onInitSuccess = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitSuccess(onInitSuccess);

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus before load");

        attributeFilterHandler.init("init");

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus during load");

        await waitForAsync();

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus after load");

        expect(onInicStart).toHaveBeenCalledTimes(1);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart parameters");
        expect(onInitSuccess).toHaveBeenCalledTimes(1);
        expect(onInitSuccess.mock.calls[0]).toMatchSnapshot("onInitSuccess parameters");
    });

    it("should trigger init error callback", async () => {
        const onInicStart = jest.fn();
        const onInitError = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("nonExisting");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitError(onInitError);

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus before load");

        attributeFilterHandler.init("init");

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus during load");

        await waitForAsync();

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus after load");

        expect(onInicStart).toHaveBeenCalledTimes(1);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart parameters");
        expect(onInitError).toHaveBeenCalledTimes(1);
        expect(onInitError.mock.calls[0]).toMatchSnapshot("onInitError parameters");
    });

    it("should trigger init cancel callback on another init call", async () => {
        const onInicStart = jest.fn();
        const onInitCancel = jest.fn();
        const attributeFilterHandler = newTestAttributeFilterHandler("positive");

        attributeFilterHandler.onInitStart(onInicStart);
        attributeFilterHandler.onInitCancel(onInitCancel);

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus before load");

        attributeFilterHandler.init("init1");

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus during load");

        attributeFilterHandler.init("init2");

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus during another load");

        await waitForAsync();

        expect(attributeFilterHandler.getInitStatus()).toMatchSnapshot("getInitStatus after load");

        expect(onInicStart).toHaveBeenCalledTimes(2);
        expect(onInicStart.mock.calls[0]).toMatchSnapshot("onInicStart 1 parameters");
        expect(onInicStart.mock.calls[1]).toMatchSnapshot("onInicStart 2 parameters");
        expect(onInitCancel).toHaveBeenCalledTimes(1);
        expect(onInitCancel.mock.calls[1]).toMatchSnapshot("onInitCancel parameters");
    });
});
