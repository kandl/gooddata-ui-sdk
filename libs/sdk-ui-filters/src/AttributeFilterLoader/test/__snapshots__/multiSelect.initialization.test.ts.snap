// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: getInitStatus after load 1`] = `"success"`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: getInitStatus before load 1`] = `"pending"`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: getInitStatus during another load 1`] = `"loading"`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: getInitStatus during load 1`] = `"loading"`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInicStart 1 parameters 1`] = `
Array [
  Object {
    "correlation": "init1",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInicStart 2 parameters 1`] = `
Array [
  Object {
    "correlation": "init2",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init cancel callback on another init call: onInitCancel parameters 1`] = `undefined`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: getInitStatus after load 1`] = `"error"`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: getInitStatus before load 1`] = `"pending"`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: getInitStatus during load 1`] = `"loading"`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: onInicStart parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init error callback: onInitError parameters 1`] = `
Array [
  Object {
    "correlation": "init",
    "error": [Error: No attribute recording non-existing-displayForm],
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: getInitStatus after load 1`] = `"success"`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: getInitStatus before load 1`] = `"pending"`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: getInitStatus during load 1`] = `"loading"`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: onInicStart parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;

exports[`MultiSelectAttributeFilterHandler should trigger init success callback: onInitSuccess parameters 1`] = `
Array [
  Object {
    "correlation": "init",
  },
]
`;
